src/main/ruby/truffleruby/core/string.rb

```diff
@@ -205,7 +205,7 @@ def rpartition(pattern)
     if pattern.kind_of? Regexp
       if m = Truffle::RegexpOperations.search_region(pattern, self, 0, size, false)
         Primitive.regexp_last_match_set(Primitive.caller_special_variables, m)
-        [m.pre_match, m[0], m.post_match]
+        return [m.pre_match, m[0], m.post_match]
```

Comment:
Hmmm was this always effectively a typo then? A missing `return`?

Reply:
could be, yes